cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -351,5 +355,278 @@ ExtensionIdRegistry* default_extension_id_registry() {
+
+  substrait_call.add_args()->CopyFrom(*value_1);
+  substrait_call.add_args()->CopyFrom(*value_2);
+  return std::move(substrait_call);
+};
+
+
+// Boolean Functions mapping
+SubstraitToArrow substrait_not_to_arrow = [] (const substrait::Expression::ScalarFunction& call) -> Result<arrow::compute::Expression>  {
+  auto value_1 = call.args(1);
+  ExtensionSet ext_set_;
+  ARROW_ASSIGN_OR_RAISE(auto expression_1, FromProto(value_1, ext_set_));
+  return arrow::compute::call("invert", {expression_1});
+};
+
+SubstraitToArrow substrait_or_to_arrow = [] (const substrait::Expression::ScalarFunction& call) -> Result<arrow::compute::Expression>  {
+  int num_args = call.args_size();  // OR function has variadic arguments
+  substrait::Expression value;
+  ExtensionSet ext_set_;
+  arrow::compute::Expression expression;
+  std::vector<arrow::compute::Expression> func_args;
+  for(int i=0; i<num_args; ++i){
+    value = call.args(i);
+    ARROW_ASSIGN_OR_RAISE(expression, FromProto(value, ext_set_));
+    func_args.push_back(expression);
+  }
+  return arrow::compute::call("or_kleene", func_args);
+};
+
+SubstraitToArrow substrait_and_to_arrow = [] (const substrait::Expression::ScalarFunction& call) -> Result<arrow::compute::Expression>  {
+  int num_args = call.args_size();  // AND function has variadic arguments
+  substrait::Expression value;
+  ExtensionSet ext_set_;
+  arrow::compute::Expression expression;
+  std::vector<arrow::compute::Expression> func_args;
+  for(int i=0; i<num_args; ++i){
+    value = call.args(i);
+    ARROW_ASSIGN_OR_RAISE(expression, FromProto(value, ext_set_));
+    func_args.push_back(expression);
+  }
+  return arrow::compute::call("and_kleene", func_args);
+};
+
+SubstraitToArrow substrait_xor_to_arrow = [] (const substrait::Expression::ScalarFunction& call) -> Result<arrow::compute::Expression>  {
+  auto value_1 = call.args(0);
+  auto value_2 = call.args(1);
+  ExtensionSet ext_set_;
+  ARROW_ASSIGN_OR_RAISE(auto expression_1, FromProto(value_1, ext_set_));
+  ARROW_ASSIGN_OR_RAISE(auto expression_2, FromProto(value_2, ext_set_));
+  return arrow::compute::call("xor", {expression_1, expression_2});
+};
+
+ArrowToSubstrait arrow_invert_to_substrait = [] (const arrow::compute::Expression::Call& call, ExtensionSet* ext_set_) -> Result<substrait::Expression::ScalarFunction> {
+  substrait::Expression::ScalarFunction substrait_call;
+  
+  ARROW_ASSIGN_OR_RAISE(auto function_reference, ext_set_->EncodeFunction("not"));
+  substrait_call.set_function_reference(function_reference);
+  
+  auto expression_1 = call.arguments[0];
+  auto expression_2 = call.arguments[1];
+  
+  ARROW_ASSIGN_OR_RAISE(auto value_1, ToProto(expression_1, ext_set_));
+  ARROW_ASSIGN_OR_RAISE(auto value_2, ToProto(expression_2, ext_set_));
+
+  substrait_call.add_args()->CopyFrom(*value_1);
+  substrait_call.add_args()->CopyFrom(*value_2);
+  return std::move(substrait_call);
+
+};
```

Comment:
Several functions map exactly and fit this simple pattern.  Can we create a helper method for:

 * Grab all arguments from Arrow call
 * Convert all arguments to Substrait expressions
 * Create Substrait call
 
Then all we should need to specify is the substrait function name.

Reply:
Made the change, add the helpers, thanks!